src/common/type/pack.c

```diff
@@ -0,0 +1,1513 @@
+/***********************************************************************************************************************************
+Pack Type
+
+Each pack field begins with a one byte tag. The four high order bits of the tag contain the field type (PackType). The four lower
+order bits vary by type.
+
+Integer types (packTypeData[type].valueMultiBit) when the value is >= -1 and <= 1:
+  3 more value indicator set to 0
+  2 value low-order bit
+  1 more ID delta indicator
+  0 ID delta low order bit
+
```

Comment:
Given the "Structure:" representation below this (from the tests), can you clarify the following:

high bits:      low bits:
3   2   1   0 | 3   2   1   0
-----------------------------  // array item 2, type u64, value 1
1   0   1   1   0   1   0   0 = B4 => B = pckTypeU64 (11), 4 =  "value of low order bit" which is 1 so what would it be if it were -1? 
-----------------------------  // id 10, type u64, value 1
1   0   1   1   0   1   1   0 = B601 => pckTypeU64 (11), 6 = "value low-order bit" + "more ID delta indicator" then, 01 according to the description "the tag will be followed by a base-128 encoded integer with the higher order ID delta bits. The ID delta represents the delta from the ID of the previous field.", the previous field ID was 7 and this field is 10 (8, 9, are missing/null) so the delta from the previous field is 3 so why is the value only 01? Since the "more ID delta indicator" seems to mean nothing here then why is this field not also represented as B4 as is the array one above?    

Structure:
           "1:u64:488"
             ", 2:u64:1911246845"
             ", 7:u64:18446744073709551615"
            **", 10:u64:1"**
            ", 11:u64:77"
            ", 12:u32:127"
            ", 13:i64:-1"
            ", 14:i32:-1"
            ", 15:bool:true"
            ", 20:bool:false"
            ", 28:obj:"
            "{"
                   "1:bool:true"
                   ", 2:bool:false"
            "}"
            ", 37:array:"
            "["
                  "1:u64:0"
                **", 2:u64:1"**
                ", 3:u64:2"
                ", 4:u64:3"
            "]"

Reply:
Hopefully clarifications in f50252aa78e9e help some here. Also, you are planning to add some examples.